pom.xml

```diff
@@ -35,6 +43,37 @@
             <artifactId>slf4j-simple</artifactId>
             <version>${slfj.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>${junit5.version}</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.avro</groupId>
+                <artifactId>avro-maven-plugin</artifactId>
+                <version>${avro.version}</version>
+                <executions>
+                    <execution>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>schema</goal>
+                            <goal>protocol</goal>
+                            <goal>idl-protocol</goal>
+                        </goals>
+                        <configuration>
+                            <sourceDirectory>${project.basedir}/src/main/avro/</sourceDirectory>
+                            <outputDirectory>${project.basedir}/src/main/java/</outputDirectory>
```

Comment:
This doesn't seems right - to put generated sources in real sources. It is easy to add them to the vcs , which is not what is desired. I guess more appropriate place is target/generated-sources .

Reply:
This will happen automatically - just remove this config section.
You will need to add proper imports then in your classes ..

Reply:
Fixed